syntax = "proto3";

package mpm;

service Processor {
    rpc ProcessModel (ModelRequest) returns (ModelResponse) {}
}

message ModelRequest {
    string seed = 1;     // env seed
    bytes model = 2;     // pickled model
    string b64array = 3; // np array data (state)
    uint32 size = 4;     // array size 
    uint32 action = 5;   // env chosen action
    int32 reward = 6;    // total reward
    double gamma = 7;    // gamma
}

message ModelResponse {
    string seed = 1;     // env seed
    bytes model = 2;     // pickled model
    string b64array = 3; // np array data (next_state)
    uint32 size = 4;     // array size
    string info = 5;     // env.step ALE info
    int32 reward = 6;    // current reward
    bool done = 7;       // done
}